bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala

```diff
@@ -496,6 +496,24 @@ case class TimeProblems(p: Proposal, s: ShellAdvisor) {
     case _ => None
   }
   lazy val b3ReqOrZero = b3Req.map(_.time).getOrElse(TimeAmount.empty)
+  lazy val jointNotAllowed = {
+    def checkForNotAllowedJointProposals(subs: Option[List[NgoSubmission]]):List[Option[Problem]] = {
+      val r = for {
+          subs <- subs
+          if subs.size > 1
+        } yield for {
+            p <- subs.filter(s => Partners.jointProposalNotAllowed.contains(s.partner))
+          } yield new Problem(Severity.Error, s"${~Partners.name.get(p.partner)} cannot be part of a joint proposal, please update the time request.", SCHEDULING_SECTION,
+              s.inPartnersView(_.editSubmissionTime(p)))
+      r.sequence
+    }
+
+    p.proposalClass match {
+      case p: GeminiNormalProposalClass => checkForNotAllowedJointProposals(p.subs.left.toOption)
+      case e: ExchangeProposalClass     => checkForNotAllowedJointProposals(e.subs.some)
+      case x                            => println("abc " + x);Nil
+    }
+  }
```

Comment:
* I'm not that familiar with how the robot works but why does the check return a `List[Option[Problem]]` instead of just a `List[Problem]`.  I think just `.flatten`ing the return value would get rid of the options.

* It may be just a question of preferences but I always find the nested `for` expressions confusing.  Since we're filtering and mapping, would `collect` be easier to follow?  Maybe something like:

````scala
val r = subs.collect {
  case Some(subList) if subList.size > 1 =>
    subList.collect {
      case sub if Partners.jointProposalNotAllowed.contains(sub.partner) =>  
        new Problem(...)
    }
}
````

* `println("abc " + x);` looks like a bit of debugging left over?